### PR TITLE
KEYCLOAK-18503 Regex Policy for authorization service

### PR DESCRIPTION
**Pull request by y-tabata**
Code PR: https://github.com/keycloak/keycloak/pull/8184

### CODE REVIEW

**Review by pedroigor — DISMISSED** [review]

**Comment by y-tabata** [comment]
@pedroigor Thanks for reviewing.
Are there any issues block this to merge?

**Comment by pedroigor** [comment]
@y-tabata I can't merge :) @andymunro can we get this one in? It is approved already. Thanks.

**Comment by y-tabata** [comment]
@andymunro Could you review this?

File: authorization_services/topics/policy-regex-policy.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
+[[_policy_regex]]
+= Regex-Based Policy
+
+You can use this type of policy to define regex conditions for your permissions.
+
+To create a new regex-based policy, select *Regex* in the dropdown list in the upper right corner of the policy listing.
```
Comment (andymunro) [inline authorization_services/topics/policy-regex-policy.adoc]: We now call it an "item list" - IBM/Red Hat standards.

**Review by andymunro — CHANGES_REQUESTED** [review]
one minor change

File: authorization_services/topics/policy-regex-policy.adoc (hunk 1 above)
Comment (y-tabata) [reply authorization_services/topics/policy-regex-policy.adoc]: @andymunro I changed "dropdown list" to "item list".

**Comment by y-tabata** [comment]
@andymunro Is there any other required change on this?

**Comment by y-tabata** [comment]
@andymunro @pedroigor Can we move forward to merge this?

**Review by pedroigor — DISMISSED** [review]

**Comment by y-tabata** [comment]
@andymunro Could you merge this?

**Comment by y-tabata** [comment]
@andymunro Could you merge this?

**Review by andymunro — DISMISSED** [review]

**Comment by pedroigor** [comment]
@mposolda Could please you merge this one?

**Review by mposolda — DISMISSED** [review]
@pedroigor sure

**Comment by mposolda** [comment]
@pedroigor @y-tabata Sorry, I've actually checked the test results and it seems to me that test failures can be related to this PR? Could you please doublecheck?

**Comment by pedroigor** [comment]
@mposolda I think this is due to the fact that in authz docs we are not using {project_images} placeholder but the relative path of the image.

It is like that everywhere in authz doc. Perhaps we should send another one with updates to docs.

**Comment by mposolda** [comment]
@pedroigor I just see that with this PR, there is regression and failing test :) Without this PR, there was no failure in AuthorizationServicesTest and with this PR, it is. So from my point of view, it is not good to merge this?

IMO we can have another PR, which will fix this and merge that one first before this PR if it is what you propose? But IMO it is not good to merge this one to have broken master.

**Comment by y-tabata** [comment]
@pedroigor @mposolda In my understanding, there is no image for RHSSO, so the test failed.

**Comment by mposolda** [comment]
@y-tabata Maybe. However I see this test fails for both in `Test / Build and test Keycloak documentation (pull_request) ` and `Test / Build and test Product documentation (pull_request)` . So likely not related only to RHSSO product, but also to Keycloak?

**Comment by y-tabata** [comment]
@mposolda I found that the image path is not correct and I fixed it. Could you review this again?

**Comment by mposolda** [comment]
@y-tabata It is better, Thanks.

So now the failures is just in the "product" build with `checkImages(org.keycloak.documentation.test.AuthorizationServicesTest)  Time elapsed: 0.005 sec  <<< FAILURE!
* ./rhsso-images/policy/create-regex.png`

Coud you try to fix it? We cannot merge PR, which introduces regression and test failures.

**Comment by y-tabata** [comment]
@mposolda 
> Coud you try to fix it? We cannot merge PR, which introduces regression and test failures.

I cannot prepare the rhsso image, so I put the image in the `images`, like the client scope policy.

**Review by pedroigor — APPROVED** [review]

**Comment by pedroigor** [comment]
@andymunro @y-tabata @mposolda Thanks.

**Pull request merged**